core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2591,14 +2598,18 @@ private void loadMetadataAndJournal(RpcContext rpcContext, LockedInodePath inode
             }
             try (TempInodePathForChild tempInodePath =
                 new TempInodePathForChild(inodePath, childStatus.getName())) {
-              DescendantType loadDescendantType =
-                  (options.getLoadDescendantType() == DescendantType.ONE) ? DescendantType.NONE :
-                      DescendantType.ALL;
               LoadMetadataOptions loadMetadataOptions =
-                  LoadMetadataOptions.defaults().setLoadDescendantType(loadDescendantType)
-                      .setCreateAncestors(false).setUfsStatus(childStatus);
+                  LoadMetadataOptions.defaults().setLoadDescendantType(DescendantType.NONE)
+                      .setCreateAncestors(true).setUfsStatus(childStatus);
               loadMetadataAndJournal(rpcContext, tempInodePath, loadMetadataOptions);
+
+              if (options.getLoadDescendantType() == DescendantType.ALL
+                  && tempInodePath.getInode().isDirectory()) {
+                InodeDirectory inodeDirectory = (InodeDirectory) tempInodePath.getInode();
+                inodeDirectory.setDirectChildrenLoaded(true);
```

Comment:
is this setting the boolean for the "child" inode?

Why don't we just set the boolean for the current inode for this method? Like, if we list the children for this inode, then we set the boolean for this inode.

Reply:
since we call the function `loadMetadataAndJournal` with `DescendantType.NONE`  for directories in the ufsStatus list, we don't actually know we are listing the status of directories once we make the recursive call.